yabt/builders/golang.py

```diff
@@ -116,71 +133,88 @@ def rm_all_but_go_mod(workspace_dir):
     packages are managed in the user machine and not inside the ephemeral
     docker.
-    When we clean the workspace we make sure to keep the go.mod since it is new
-    go build redownload all packages (can we solve this?)
-
 
     TODOs:
-      - "replace proto => ./proto" is needed since the generated code import
-        doesn't have the package before imports of other generated go files.
-        See if there is another way to do it (understanding that can help us
-        create a GoLib builder)
+      - Support multiple modules in the same repo. Currently only one module
+        is supported. Better than to define go_module in YSettings and not in
+        the different targets. Using different go_modules is untested, and
+        will probably will not be able to import packages from one module in
+        another one.
+
     """
+
     builder_name = target.builder_name
     yprint(build_context.conf, command, builder_name, target)
-    workspace_dir = build_context.get_workspace(builder_name, target.name)
-    go_package = (target.props.get('go_package') or
-                  build_context.conf.get('go_package', None))
-    go_mod_path = join(workspace_dir, 'go.mod')
-    if not go_package:
-        raise KeyError('Must specify go_package in {} common_conf '
-                       'or on target'.format(YSETTINGS_FILE))
 
-    # we leave the go.mod file otherwise the caching of downloaded packages
-    # doesn't work
+    go_module = build_context.conf.get('go_module', None)
+    if not go_module:
+        raise KeyError("Must specify go_module in {} common_conf".format(
+            YSETTINGS_FILE
+        ))
+
+    workspace_dir = build_context.get_workspace(builder_name, target.name)
     rm_all_but_go_mod(workspace_dir)
-    binary = join(*split(target.name))
 
     buildenv_workspace = build_context.conf.host_to_buildenv_path(
-        workspace_dir)
-    buildenv_sources = [join(buildenv_workspace, src)
-                        for src in target.props.sources]
-    if target.props.get('mod_file'):
-        link_node(join(build_context.conf.project_root,
-                       target.props.get('mod_file')),
-                  go_mod_path)
-    sources_to_link = list(target.props.sources)
-    has_protos = False
-
-    # Goging over all deps and pulling their sources.
-    # TODO(eyal): This pull sources of all types which is most likely not
-    # needed but not harming. We should revisit this in later iterations.
+        workspace_dir
+    )
+    buildenv_sources = [
+        join(buildenv_workspace, src) for src in target.props.sources
+    ]
+
+    files_to_link = list(target.props.sources)
+
     for dep in build_context.generate_all_deps(target):
-        sources_to_link.extend(dep.props.get('sources', []))
+        files_to_link.extend(filter(lambda x: x.endswith('.go'),
+                                    dep.props.get('sources', [])))
+        files_to_link.extend(dep.props.get('files', []))
```

Comment:
that's a bit dangerous as you can't be sure which builders will define a 'files' prop, maybe limit to builders of type FilesGroup?